utils/ccdatofhir/mapping_resource_immunization.bal

```diff
@@ -0,0 +1,262 @@
+        r4:Immunization immunization = {patient: {}, occurrenceDateTime: "", occurrenceString: "", vaccineCode: {}, status: "not-done"};
+
+        xml idElement = substanceAdministrationElement/<v3:id|id>;
+        xml statusCodeElement = substanceAdministrationElement/<v3:statusCode|statusCode>;
+        xml effectiveTimeElement = substanceAdministrationElement/<v3:effectiveTime|effectiveTime>;
+        xml repeatNumberElement = substanceAdministrationElement/<v3:repeatNumber|repeatNumber>;
+        xml routeCodeElement = substanceAdministrationElement/<v3:routeCode|routeCode>;
+        xml approachSiteCodeElement = substanceAdministrationElement/<v3:approachSiteCode|approachSiteCode>;
+        xml doseQuantityElement = substanceAdministrationElement/<v3:doseQuantity|doseQuantity>;
+        xml vaccineCodeElement = substanceAdministrationElement/<v3:consumable|consumable>/<v3:manufacturedProduct|manufacturedProduct>/<v3:manufacturedMaterial|manufacturedMaterial>/<v3:code|code>;
+        xml lotNumberTextElement = substanceAdministrationElement/<v3:consumable|consumable>/<v3:manufacturedProduct|manufacturedProduct>/<v3:manufacturedMaterial|manufacturedMaterial>/<v3:lotNumberText|lotNumberText>;
+        xml performerElement = substanceAdministrationElement/<v3:performer|performer>;
+        xml entryRelationship = substanceAdministrationElement/<v3:entryRelationship|entryRelationship>;
+
+        int index = 0;
+        foreach xml idElem in idElement {
+            r4:Identifier? mapCcdaIdToFhirIdentifierResult = mapCcdaIdToFhirIdentifier(idElem);
+            if mapCcdaIdToFhirIdentifierResult is r4:Identifier {
+                immunization.identifier[index] = mapCcdaIdToFhirIdentifierResult;
+                index = index + 1;
+            }
+        }
+
+        immunization.status = mapCcdaStatusCodeToFhirImmunizationStatus(statusCodeElement);
+
+        r4:dateTime? mapCCDAEffectiveTimetoFHIRDateTimeResult = mapCcdaDateTimeToFhirDateTime(effectiveTimeElement);
+        if mapCCDAEffectiveTimetoFHIRDateTimeResult is r4:dateTime {
+            immunization.occurrenceDateTime = mapCCDAEffectiveTimetoFHIRDateTimeResult;
+        }
+
+        string|error? repeatNumber = repeatNumberElement.value;
+        if repeatNumber is string && repeatNumber != "" {
+            int|error repeatNumberValue = int:fromString(repeatNumber);
+            if repeatNumberValue is int {
+                immunization.protocolApplied = [{doseNumberString: repeatNumber, doseNumberPositiveInt: repeatNumberValue}];
+            } else {
+                log:printDebug("Repeat Number is not available", repeatNumberValue);
+            }
+        } else {
+            log:printDebug("Repeat Number is not available");
+        }
+
+        immunization.route = mapCcdaCodingtoFhirCodeableConcept(routeCodeElement);
+        immunization.site = mapCcdaCodingtoFhirCodeableConcept(approachSiteCodeElement);
+
+        string|error? doseQuantity = doseQuantityElement.unit;
```

Comment:
```suggestion
        string|error doseQuantity = doseQuantityElement.unit;
```
Check all the places where you have unnecessary `?`

Reply:
https://discord.com/channels/957996897782616114/1123904724425523210/1124169449356931072